### PR TITLE
MultiMap test enhancement

### PR DESCRIPTION
**Pull request by zhostasa**
* minor enhancements in MultiMap tests

### CODE REVIEW

File: multimap/src/test/java/org/infinispan/multimap/impl/SuperUser.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+import static java.util.Objects.hash;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+
+import org.infinispan.commons.marshall.SerializeWith;
+
+@SerializeWith(SuperUser.SuperUserExternalizer.class)
+public class SuperUser extends User {
```
Comment (karesti) [inline multimap/src/test/java/org/infinispan/multimap/impl/SuperUser.java]: The comment I got in JDG was that I created a User, but in reality we not using the "age" in the tests. So there is already a bean in tests called "Person" that serves the same purpose. 
Would you mind to do the modification in the PR ? It's not a bug deal, I would do it when this branch will be integrated in infinispan -hopefully soon :)

File: multimap/src/test/java/org/infinispan/multimap/impl/DistributedMultimapCacheTest.java [hunk 2]
```diff
@@ -288,6 +409,58 @@ public void testGetEntry() {
...
                   )
       );
+      
+      await(multimapCache.put(EMPTY_KEY, RAMON).thenCompose(r3 -> multimapCache.getEntry(EMPTY_KEY)).thenAccept(v -> {
+         assertTrue(v.isPresent() && v.get().getKey().equals(EMPTY_KEY) && v.get().getValue().contains(RAMON));
+      }));
+   }
+   
+   public void testPutWithNull() {
+      MultimapCache<String, User> multimapCache = getMultimapCacheMember(NAMES_KEY);
```
Comment (karesti) [inline multimap/src/test/java/org/infinispan/multimap/impl/DistributedMultimapCacheTest.java]: aren't these null tests tests already covered ?

**Review by karesti — COMMENTED** [review]
Some questions concerning null key/values tests and User/Person bean ! thanks for the PR !

File: multimap/src/test/java/org/infinispan/multimap/impl/DistributedMultimapCacheTest.java (hunk 2 above)
Comment (zhostasa) [reply multimap/src/test/java/org/infinispan/multimap/impl/DistributedMultimapCacheTest.java]: They were for Embedded, not for Distributed (which is sad as I wrote these before I moved on to Embedded, then found them there, so I removed mine and just copied these)

File: multimap/src/test/java/org/infinispan/multimap/impl/SuperUser.java (hunk 1 above)
Comment (zhostasa) [reply multimap/src/test/java/org/infinispan/multimap/impl/SuperUser.java]: Right, forgot about that in the morning stupor, swapped

**Review by karesti — APPROVED** [review]

**Comment by karesti** [comment]
integrated, thanks

**Pull request merged**